ui/src/table.tsx

```diff
@@ -733,16 +733,17 @@ export const
           groupedBy: Dict<any> = []
 
         if (m.groups) {
-          groups = filteredItems.reduce((acc, { group }, idx) => {
-            const prevGroup = acc[acc.length - 1]
-            prevGroup?.key === group
-              ? prevGroup.count++
-              : acc.push({ key: group, name: group, startIndex: idx, count: 1, isCollapsed: getIsCollapsed(group, expandedRefs.current) })
+          groups = m.groups.reduce((acc, { label }) => {
+            const
+              prevGroup = acc[acc.length - 1],
+              startIndex = prevGroup ? prevGroup.startIndex + prevGroup.count : 0,
+              count = filteredItems.filter(({ group }) => group === label).length
+            acc.push({ key: label, name: label, startIndex, count, isCollapsed: getIsCollapsed(label, expandedRefs.current) })
```

Comment:
This is slow. I wasn't able to figure out why this change is needed either. Can you explain?

Reply:
> I wasn't able to figure out why this change is needed either.

With previous implementation `groups` were created based on `filteredItems` which may not contain all available groups from `m.groups`.

Reply:
I see. However, this approach is not good as it yields O(N^2) compared to the original O(N).